ports/nrf/common-hal/neopixel_write/__init__.c

```diff
@@ -132,16 +134,29 @@ void common_hal_neopixel_write (const digitalio_digitalinout_obj_t* digitalinout
         } else {
             uint8_t sd_en = 0;
             (void) sd_softdevice_is_enabled(&sd_en);
-            if (sd_en) {
-                // If the soft device is enabled then we must use PWM to
-                // transmit. This takes a bunch of memory to do so raise an
-                // exception if we can't.
-                pixels_pattern = (uint16_t *) m_malloc(pattern_size, false);
-            } else {
-                pixels_pattern = (uint16_t *) m_malloc_maybe(pattern_size, false);
-            }
 
-            pattern_on_heap = true;
+            if (!pattern_on_heap || pixels_pattern_heap_size < pattern_size) {
+                if (pattern_on_heap) {
```

Comment:
```suggestion
                if (pixels_pattern_heap_size < pattern_size) {
```

Reply:
Fixed the submodule and addressed this review.